spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AwsS3EnvironmentRepositoryFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.config.server.environment;
+
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+import org.springframework.cloud.config.server.config.ConfigServerProperties;
+
+public class AwsS3EnvironmentRepositoryFactory implements
+		EnvironmentRepositoryFactory<AwsS3EnvironmentRepository, AwsS3EnvironmentProperties> {
+
+	final private ConfigServerProperties server;
+
+	public AwsS3EnvironmentRepositoryFactory(ConfigServerProperties server) {
+		this.server = server;
+	}
+
+	@Override
+	public AwsS3EnvironmentRepository build(
+			AwsS3EnvironmentProperties environmentProperties) {
+		final AmazonS3ClientBuilder clientBuilder = AmazonS3ClientBuilder.standard();
```

Comment:
I believe we should add the ability to override the baseUrl of the s3 client. This will allow support for beta regions of S3, and other s3 compatible storage APIs (GCS, Azure CS)

https://stackoverflow.com/questions/45055921/how-to-set-base-url-in-java-aws-sdk

Reply:
fixed